### ISSUE TITLE
repurpose unused eye slot on skin pngs as afk / crying eye

### ISSUE THREAD

**Issue by l-ouis** [ddnet/ddnet#9423]
![image](https://github.com/user-attachments/assets/f24b199d-994f-4c96-99d9-1580c4021c05)

Circled is a + eye that is never used afaik.

My proposal is to instead use this for the afk / crying eyes so newer skins can have custom graphics for these eyes instead of using the smushed default eyes.

Cons:
Need to replace every skin in the db to replace these eyes with scaled normal eyes using some script
Have to manage people's custom skins getting broken

**Comment by l-ouis** [comment]
If this is considered then we should also fix the foot scaling and instead use a script to "re-scale" all the feet in the DB and just use 1x scaling while rendering ingame. Currently feet are upscaled ingame which makes them a lil more blurry than body. It's also frustrating for skin creators since the scaling isn't obvious inside the png

**Comment by AssassinTee** [comment]
I just go with "accepting patches here", I don't think adding and maintaining yet another eye emoji is too difficult.

I propose 
1. do the client and server change
2. Provide a way to upgrade your own skins
3. Then updating all the skins in the database
I am worried about the second step more, to be honest. @SoulyVEVO as maintainer (are you maintainer or just database crew?) of the skin database, would this be fine for you?

**Comment by SoulyVEVO** [comment]
@AssassinTee looks good to me

**Comment by Patiga** [comment]
> If this is considered then we should also fix the foot scaling by using a script to "re-scale" all the feet in the DB and just use 1x scaling while rendering ingame. Currently feet are upscaled ingame which makes them a lil more blurry than body. It's also frustrating for skin creators since you cant visualize the scaling while editing the skin image

This should be a separate issue

**Comment by Patiga** [comment]
I don't feel like the sleeping eyes + crying emoji eyes are semantically close. Wouldn't it make more sense to stay with the sleeping eyes as-is, and make it possible to customize the crying eyes?
Updating all skins is quite a big endeavor (not to forget breaking all old skins), and should have substantial motivation. I think the idea is neat, but could you maybe provide some examples images of how it looks better with custom eyes?

**Comment by AssassinTee** [comment]
> Wouldn't it make more sense to stay with the sleeping eyes as-is, and make it possible to customize the crying eyes?

I don't get this point, the crying eyes are not a substitute for the sleepy eyes, we have an **empty/unused slot**, which is known as `dead eyes`. For 0.7 skins I made it fallback to the pain-eyes, because I deem them semantically closest.

> Updating all skins is quite a big endeavor (not to forget breaking all old skins)

We are not "breaking" old skins, the new emote (only the new emote) may look weird because the wrong eye sprite is used if you don't update your own skins. I think I can live with that, but that's why we have point 2 here

`Provide a way to upgrade your own skins` to allow people to migrate their local skins

>  I think the idea is neat, but could you maybe provide some examples images of how it looks better with custom eyes?

@l-ouis I think he's right, we should provide an example of crying eyes 😆 If you have a better idea, than a crying eye, go ahead ^^ renaming this is easy at this stage, an afk eye or a sunglasses-eye would also work

**Comment by Patiga** [comment]
> > Wouldn't it make more sense to stay with the sleeping eyes as-is, and make it possible to customize the crying eyes?
> 
> I don't get this point, the crying eyes are not a substitute for the sleepy eyes, we have an **empty/unused slot**, which is known as `dead eyes`. For 0.7 skins I made it fallback to the pain-eyes, because I deem them semantically closest.

I might've misunderstood
> My proposal is to instead use this for the afk / crying eyes so newer skins can have custom graphics for these eyes instead of using the smushed default eyes.

My question was why the new slot should be used for both afk eyes **and** crying eyes at the same time. That's the intended route, no? Crying eyes on afk tees doesn't sound fitting.

> We are not "breaking" old skins, the new emote (only the new emote) may look weird because the wrong eye sprite is used if you don't update your own skins. I think I can live with that, but that's why we have point 2 here
> `Provide a way to upgrade your own skins` to allow people to migrate their local skins

A lot of existing skins are missing that slot, meaning that those skins will have no visible eyes on such emotes. I'd strongly consider that breaking skins. (e.g. even official skins in the repo like `chinese_by_whis`, `beast`, but mostly db skins where the eyes were modified such as many skins from the creators `00` or `+KZ`)
For that reason, it is *required* to provide a way to update, which even with update scripts is a significant effort. Proper motivation should validate that effort.

**Comment by AssassinTee** [comment]
> My question was why the new slot should be used for both afk eyes and crying eyes at the same time. That's the intended route, no? Crying eyes on afk tees doesn't sound fitting.

My current draft only adds **one** emote (with the same size as the other ones)

> For that reason, it is required to provide a way to update, which even with update scripts is a significant effort. Proper motivation should validate that effort.

yes, fully agreed. The update script should be able to force a new default sprite or do this softly, only adding this sprite if the slot is empty. I wonder if we should add a tool for this

Even if you run around with a skin, where this slot is empty, nobody either sees your skin (because it's not in the db), you choose and you chose to run around eyeless yourself (which you can already do by modifying any emote)

**Comment by l-ouis** [comment]
Oops, maybe i worded the initial issue badly. I just wanted the new slot to be used for "closed eyes".  I have a script that replaces a skin png with the previous "closed eyes", which is just a squished version of the default eyes.

The closed eyes are also used for the crying emote, which i think can stay.

Squishing the default eyes on some skins is ugly, since the eyes on some skins also include a mouth / nose. see the pig skin, the robot skin, etc. for reference

<img width="2880" height="1800" alt="Image" src="https://github.com/user-attachments/assets/d78c09b9-d6ce-4ee0-a50b-ddc186d1c705" />

<img width="2880" height="1800" alt="Image" src="https://github.com/user-attachments/assets/72682af8-9c8e-4941-a315-9252a0704cad" />

**Comment by l-ouis** [comment]
yes, it breaks any skins that the player has saved in their data folder, which is not good. However it is easy to host some online webtool that converts these skins to the new format, and the skindb and shipped skins can be updated easily. I wrote a quick 50 line python script that seems to do the job
